yt_dlp/extractor/dailymotion.py

```diff
@@ -94,10 +94,10 @@ class DailymotionIE(DailymotionBaseInfoExtractor):
     _VALID_URL = r'''(?ix)
                     https?://
                         (?:
-                            (?:(?:www|touch)\.)?dailymotion\.[a-z]{2,3}/(?:(?:(?:embed|swf|\#)/)?video|swf)|
+                            (?:(?:www|touch|geo)\.)?dailymotion\.[a-z]{2,3}/(?:(?:(?:(?:embed|swf|\#)/)|player\.html\?)?video|swf)|
                             (?:www\.)?lequipe\.fr/video
                         )
-                        /(?P<id>[^/?_]+)(?:.+?\bplaylist=(?P<playlist_id>x[0-9a-z]+))?
+                        (?:\/|\=)(?P<id>[^/?_&]+)(?:.+?\bplaylist=(?P<playlist_id>x[0-9a-z]+))?
```

Comment:
```suggestion
                        [/=](?P<id>[^/?_&]+)(?:.+?\bplaylist=(?P<playlist_id>x[0-9a-z]+))?
```

Reply:
I thought about it and a tested again with some another case and i think its ok. If you think its not good, i can move geo.dailymotion with different pattern and different class.

Reply:
I assume u r replying to:

> By trying to capture 2 very different patterns with one regex, there is a much higher possibility of encountering false positives.

Just wanted to make you aware of the risk. If you think it's a non-issue, it's fine

But this suggestion is unrelated to that comment